indico/modules/designer/templates/_list.html

```diff
@@ -18,6 +18,9 @@
             {% trans %}Print badges{% endtrans %}
         {%- endif -%}
     {%- endset %}
+    {% set extra_params %}
+        {{ template_hook('badge-form-button-extra-params', tpl=tpl, regform=regform) }}
+    {% endset %}
```

Comment:
With my suggestion I meant getting rid of the `{% set %}` - you can just call the template hook in the place where you need it.

Also, you can actually return a dict of attributes from your template hook if you use it like this:

```
    {% for attrs in template_hook('xxx', as_list=true, ...) %}
        {{ attrs | html_params }}
    {% endfor %}
```

Reply:
Like that, I'd need to call the template hook twice below since I'm passing it in 2 places. Is this okay?

Reply:
Sure. Only one of the calls will happen at any time anyway.

Reply:
Just wondering, are you still going to update this to use the loop?

Reply:
No, since I don't need to return a dict of attributes.

Reply:
Wouldn't it be cleaner? When registering the template hook, instead of having to generate a string of HTML attributes (which implies creating another Jinja template in your plugin), you could simply return the dict from Python code.

Reply:
Oh right, indeed I was creating another jinja template. Let me test this way :)

Reply:
Changed the hook to use the loop